// Copyright 2020 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto2";

package feedwire;

import "components/feed/core/proto/v2/wire/action_payload.proto";
import "components/feed/core/proto/v2/wire/action_surface.proto";

option optimize_for = LITE_RUNTIME;

message FeedAction {
  message ClientData {
    enum ActionOrigin {
      UNKNOWN_ORIGIN = 0;
      CLIENT_SESSION = 1;
      CLIENT_CACHE = 2;
    }
    optional int64 timestamp_seconds = 1;
    optional int64 sequence_number = 2;
    optional int64 duration_ms = 3;
    optional ActionOrigin action_origin = 4;
    optional ActionSurface action_surface = 5;
  }
  optional ActionPayload action_payload = 2;
  optional ClientData client_data = 3;
}
